core/designsystem/src/commonMain/kotlin/com/droidknights/app/core/designsystem/components/Icon.kt

```diff
@@ -0,0 +1,62 @@
+package com.droidknights.app.core.designsystem.components
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.paint
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.toolingGraphicsLayer
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import com.droidknights.app.core.designsystem.theme.LocalContentColor
+
+// material3 - Icon
+@Composable
+fun Icon(
+    painter: Painter,
+    contentDescription: String?,
+    modifier: Modifier = Modifier,
+    tint: Color = LocalContentColor.current
+) {
+    val colorFilter =
+        remember(tint) { if (tint == Color.Unspecified) null else ColorFilter.tint(tint) }
```

Comment:
이거 recomposition에 큰 영향을 안 끼치는거라면 remember가 불필요할 것 같은데 어떻게 생각하시나요? 위의 로직이 그렇게 큰 연산을 하지는 않는다고 생각을 해서요.

```suggestion
    val colorFilter = if (tint == Color.Unspecified) null else ColorFilter.tint(tint)
```

Reply:
@l2hyunwoo 일단 구현 자체는 m3의 Icon composable 그대로인데, `ColorFilter` 생성 비용을 아끼려고 한 의도 아니었을까요?

Reply:
tint가 생성함수였군요…! 리멤버 걸어주는게 좋겠네열.